indexer/src/providers/providerStatusProvider.ts

```diff
@@ -28,103 +32,160 @@ export async function syncProvidersInfo() {
-        goVersion: string;
-        compiler: string;
-        platform: string;
-      };
-    };
+
+    await Provider.update(
+      {
+        lastSnapshotId: createdSnapshot.id,
+        isOnline: !!providerStatus,
+        error: error,
+        lastCheckDate: checkDate,
+        cosmosSdkVersion: cosmosVersion,
+        akashVersion: akashVersion,
+        deploymentCount: providerStatus?.resources.deploymentCount,
+        leaseCount: providerStatus?.resources.leaseCount,
+        activeCPU: providerStatus?.resources.activeCPU,
+        activeGPU: providerStatus?.resources.activeGPU,
+        activeMemory: providerStatus?.resources.activeMemory,
+        activeStorage: providerStatus?.resources.activeStorage,
+        pendingCPU: providerStatus?.resources.pendingCPU,
+        pendingGPU: providerStatus?.resources.pendingGPU,
+        pendingMemory: providerStatus?.resources.pendingMemory,
+        pendingStorage: providerStatus?.resources.pendingStorage,
+        availableCPU: providerStatus?.resources.availableCPU,
+        availableGPU: providerStatus?.resources.availableGPU,
+        availableMemory: providerStatus?.resources.availableMemory,
+        availableStorage: providerStatus?.resources.availableStorage
+      },
+      {
+        where: { owner: provider.owner },
+        transaction: t
+      }
+    );
+
+    if (providerStatus) {
+      for (const node of providerStatus.nodes) {
+        const providerSnapshotNode = await ProviderSnapshotNode.create(
+          {
+            snapshotId: createdSnapshot.id,
+            name: node.name,
+            cpuAllocatable: node.cpuAllocatable,
+            cpuAllocated: node.cpuAllocated,
+            memoryAllocatable: node.memoryAllocatable,
+            memoryAllocated: node.memoryAllocated,
+            ephemeralStorageAllocatable: node.ephemeralStorageAllocatable,
+            ephemeralStorageAllocated: node.ephemeralStorageAllocated,
+            capabilitiesStorageHDD: node.capabilitiesStorageHDD,
+            capabilitiesStorageSSD: node.capabilitiesStorageSSD,
+            capabilitiesStorageNVME: node.capabilitiesStorageNVME,
+            gpuAllocatable: node.gpuAllocatable,
+            gpuAllocated: node.gpuAllocated
+          },
+          { transaction: t }
+        );
+
+        await ProviderSnapshotNodeCPU.bulkCreate(
+          node.cpus.map((cpuInfo) => ({
+            snapshotNodeId: providerSnapshotNode.id,
+            vendor: cpuInfo.vendor,
+            model: cpuInfo.model,
+            vcores: cpuInfo.vcores
+          })),
+          { transaction: t }
+        );
+
+        await ProviderSnapshotNodeGPU.bulkCreate(
+          node.gpus.map((gpuInfo) => ({
+            snapshotNodeId: providerSnapshotNode.id,
+            vendor: gpuInfo.vendor,
+            name: gpuInfo.name,
+            modelId: gpuInfo.modelId,
+            interface: gpuInfo.interface,
+            memorySize: gpuInfo.memorySize // TODO: Change type to bytes?
```

Comment:
is it supposed to be bytes?

Reply:
It's fine to keep it as a string I think. Only use for a numerical type would be to query GPUs with X vram "or more", but for now it's simpler like this. I removed the todo, we can always change the type in the futur if needed.